packages/react/src/format.ts

```diff
@@ -105,4 +106,14 @@ function getElements(parts) {
   )
 }
 
+function uniqueId(prefix: "$lingui$" = "$lingui$") {
+  if (!idCounter[prefix]) {
+    idCounter[prefix] = 0
+  }
+
+  const id =++idCounter[prefix]
+
+  return `${prefix}_${id}`
+}
+
```

Comment:
I would prefer simple closure before global variable:

```suggestion
const makeCounter = (count = 0, prefix = "") => () => `${prefix}${count++}`
```

Then we can create a counter inside `formatFunction`:

```js
const uniqueId = makeCounter(0, '$lingui$')
```

I think it should be safe to create the counter on every call of `formatElements`. The problem was that elements without an index (variables) were getting the same id.

Reply:
Yes, absolutly, i've changed to match your suggestion!!